addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/internal/AuthenticationStep.java

```diff
@@ -181,6 +182,16 @@ public void setOrder(int order) {
     }
 
     public WebElement execute(WebDriver wd, UsernamePasswordAuthenticationCredentials credentials) {
+        if (getType() == Type.WAIT) {
```

Comment:
Can getType ever be null? Should the left/right switch?

Reply:
Same for below at 288

Reply:
We are validating that the type is set. Not sure I follow the second question, why switch them?

Reply:
If it could have been null